.github/workflows/benchmarking-regression.yml

```diff
@@ -0,0 +1,121 @@
+name: Performance regression tests
+
+on:
+  pull_request:
+    branches: [main, feature/lc/benchmarking_tests]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  actions: write
+  pull-requests: write
+
+run-name: Performance regression tests from ${{ github.ref }}
+
+jobs:
+  benchmarking:
+    name: Benchmarking
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest]
+        python: ["3.10"]
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Set up Python 3.x
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-pip-wheels
+        uses: actions/cache@v4
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+        
+      - uses: actions/checkout@v4
+        with:
+          ref: feature/lc/benchmarking_tests
+
+      - name: Install dependancies
+        run: poetry install --sync
+
+      - name: Benchmarking on main
+        run: poetry run pytest --benchmark-only --benchmark-save="benchmark"
+
+      - uses: actions/checkout@v4
+        with:
+          clean: false
+
+      - name: Install dependancies
+        run: 
+          poetry install --sync
+
+      - name: Benchmarking on branch
+        run: poetry run pytest --benchmark-only --benchmark-save="benchmark" --benchmark-compare --benchmark-compare-fail=min:50%
+
+      - name: Generate report 
+        id: generate-report
+        if: always()
+        shell: bash
+        run: |
+          poetry run python tests/benchmarking/generate_report.py
+          SUMMARY=$(cat .benchmarks/summary.md)
+          echo "$SUMMARY" >> $GITHUB_STEP_SUMMARY
+          {
+            echo 'markdown-summary<<EOF'
+            echo "$SUMMARY"
+            echo EOF
+          } >> "$GITHUB_OUTPUT"
+        
+      - name: Update PR
```

Comment:
Not a fan of this step: can foresee errors if we add future commented reports (e.g. test coverage, unit test report)

Reply:
added an additional check to ensure the correct comment is selected